packages/jupyter-chat/src/footers/types.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { IChatModel } from '../model';
+import { IChatMessage } from '../types';
+
+/**
+ * The properties sent to any footer component.
+ */
+export interface IChatMessageFooterProps {
+  model: IChatModel;
+  message: IChatMessage;
+}
+
+/**
+ * The chat footer added to the registry.
+ */
+export interface IChatMessageFooter {
+  component: React.FC<IChatMessageFooterProps>;
+  position: 'left' | 'center' | 'right';
+}
+
+/**
+ * The footers returned by the registry.
+ */
+export type Footers = {
+  left?: IChatMessageFooter;
+  center?: IChatMessageFooter;
+  right?: IChatMessageFooter;
+};
```

Comment:
Only one comment: I think we can improve these names to be more clear. Right now, "Footer" refers to both the group and a individual within that group.

A footer is everything at the bottom of some parent text, so it's more accurate to refer to the group of 3 as "the footer", and refer to each individual as a "footer section". I also think we can drop the prefix "IChat" from these names, since "message" almost always refers to "chat message" in this project, and interfaces don't need to be named explicitly.

```suggestion
/**
 * The props sent passed to each `MessageFooterSection` React component.
 */
export type MessageFooterSectionProps = {
  model: IChatModel;
  message: IChatMessage;
}

/**
 * A message footer section which can be added to the footer registry.
 */
export type MessageFooterSection = {
  component: React.FC<MessageFooterSectionProps>;
  position: 'left' | 'center' | 'right';
}

/**
 * The message footer returned by the registry, composed of 'left', 'center',
 * and 'right' sections.
 */
export type MessageFooter = {
  left?: MessageFooterSection;
  center?: MessageFooterSection;
  right?: MessageFooterSection;
};
```